Reset Group not working for large groupings

zakgilbert opened the pull request:
# Ticket Link

[GROUPINGS-1330](https://uhawaii.atlassian.net/browse/GROUPINGS-1330)

# List of squashed commits
- Implement resetIncludeGroup and resetExcludeGroup
- Implement modal behavoir for new reset.
- Comment code.
- Rename endpoints to adhere to REST design
- Cover endpoint on unit tests.
- Get await to work.
- Base resetModal message off of actual results.
- Refactor resetIncludeGroup and resetExcludeGroup to a single function
- Implement error handlers.
- Provide jasmine coverage for new resetGroup methods.

# Test Checklist
- [ ] Unit Tests Passed:
- [ ] Jasmine Tests Passed:
- [ ] General Visual Inspection:

zakgilbert commented:
![image](https://user-images.githubusercontent.com/26912258/216447367-e637ba02-c728-4c34-b3fc-f713253aaa13.png)
![image](https://user-images.githubusercontent.com/26912258/216447633-9ea2e814-b02d-4e9b-907f-fd83f7dd7907.png)

mhodgesatuh commented:
Jordan will be exploring a technique for addressing these cross-site scripting warning.  For now we can let this pass, but will likely have to revisit.

mhodgesatuh reviewed (approved):
I don't see any issues

JorWo reviewed (approved):
Thanks for fixing the formatting of other functions too.

zakgilbert commented:
Implemented dynamic error messages.
Renamed some modal functions and files.
Refined the jasmine tests.
Fixed the lagging spinner.

Test reflects the new messages.
![image](https://user-images.githubusercontent.com/26912258/217107003-5a8b191a-7a69-4b4b-a2f0-8a1796ce64d0.png)

JorWo commented on src/main/resources/static/javascript/mainApp/general.controller.js:
```diff
@@ -1476,11 +1474,51 @@
+                "excludeSuccess": checkResetResult(groupingPath, resetExcludeResult)
+            };
+            if (results.includeFailure || results.excludeFailure) {
+                for (let i = 0; i < Message.ResetGroupError.ResetGroupErrorMessageMap.length; i++) {
+                    let object = Message.ResetGroupError.ResetGroupErrorMessageMap[parseInt(i, 10)];
+                    if (Utility.compareObjects(object, results)) {
+                        let message = Message.ResetGroupError.ResetGroupErrorMessages.Body[parseInt(i, 10)];
```
You can use a for of loop here to avoid using parseInt(i, 10) to fix the Codacy issue.
